### PR TITLE
531 - Improve NodeModal.vue and Tapestry.vue code structures

### LINKED ISSUE #531: Improve NodeModal.vue and Tapestry.vue code structures

### PR DESCRIPTION
**Pull request by narendrasss**
Closes #531

**Changes**

* Completely move node submission logic into `NodeModal`
* Restructure submission logic to be more concise
* Restructure `NodeModal` component so that each section of the modal belongs in its own component
  * This also removes a lot of state from `NodeModal` that was really only being used in one particular part of the form
* Update `NodeModal` API to be shorter and more readable
  * It now accepts a `node-id` and `modal-type` as props and emits `submit` and `cancel` events
* Add loading states to the modal on start and on submission
* [Fix node values not reverting when a user cancels an edit ](https://app.asana.com/0/1176098622515689/1148603786212633)

### CODE REVIEW

**Comment by liam-armstrong** [comment]
Node creation works great, switching between components in the modal is very smooth.

File: templates/vue/src/components/node-modal/ContentForm.vue [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        { value: "activity", text: "Activity" },
+        { value: "accordion", text: "Accordion" },
+      ],
+    }
+  },
+  computed: {
+    ...mapGetters(["getDirectChildren", "getDirectParents", "getNode"]),
+    activeForm() {
+      return this.node.mediaType ? this.node.mediaType + "-form" : null
+    },
```
Comment (wynnset) [inline templates/vue/src/components/node-modal/ContentForm.vue:84]: Genius

**Comment by wynnset** [comment]
@narendrasss great job with this. Looks so much better! Can you please fix the merge conflicts and then mark it as ready to merge? We also need to create a TYDE PR for this and also a lot of the changes here will need to be reflected in the other outstanding PRs.

**Comment by wynnset** [comment]
Sorry @narendrasss I was going to merge this in now, but there are conflicts. Can you fix the conflicts when you are working?

**Pull request merged**